contracts/oracle/GasOracle.sol

```diff
@@ -31,26 +32,34 @@ contract GasOracle is IOracle, Ownable {
      * @dev Returned value is USD/Gas * 10^18 for compatibility with rest of calculations
      */
     function latestAnswer() external view override returns (uint256) {
-        uint256 gasPrice = uint256(gasOracle.latestAnswer());
-        uint256 ethPrice = uint256(priceOracle.latestAnswer());
+        uint256 gasPrice = _gasPriceToWad();
+        uint256 ethPrice = _ethPriceToWad();
 
         uint256 result = PRBMathUD60x18.mul(gasPrice, ethPrice);
         return result;
     }
 
     /**
-     * @notice converts a raw value to a WAD value.
-     * @dev this allows consistency for oracles used throughout the protocol
-     *      and allows oracles to have their decimals changed withou affecting
-     *      the market itself
+     * @notice Returns the latest answer from the Fast Gas/Gwei oracle in WAD format (Gwei/Gas * 10^18).
+     * @dev Since the answer is provided in Gwei, the number is scaled to 9 decimals to convert to WAD.
      */
-    function toWad(uint256 raw, IChainlinkOracle _oracle) internal view returns (uint256) {
-        IChainlinkOracle oracle = IChainlinkOracle(_oracle);
-        // reset the scaler for consistency
-        uint8 _decimals = oracle.decimals(); // 9
+    function _gasPriceToWad() internal view returns (uint256) {
+        uint256 result = uint256(gasOracle.latestAnswer());
+        uint8 _decimals = gasOracle.decimals();
+        require(_decimals <= MAX_GWEI_DECIMALS, "GAS: too many decimals");
+        uint256 scaler = uint256(10**(MAX_GWEI_DECIMALS - _decimals));
+        return result * scaler;
+    }
```

Comment:
Do we want an ability to take gas oracles that use higher precision than 9 decimal places?

e.g. a gas oracle that has 18 decimals, in which case we would want to divide by 10^9

Reply:
I was considering this, but since the oracle reports in GWEI, would it even be possible for an oracle to have a higher precision than 10^18?

Reply:
Yeah I don't think so.